### ISSUE TITLE
Should ockam_transport_core crate be merged into ockam_core?

### ISSUE THREAD

**Issue by mrinalwadhwa** [build-trust/ockam#2171]
The only thing it does is define TransportError

https://github.com/ockam-network/ockam/blob/7ec20f6ee2815b630178f4d372869cbb99d000e8/implementations/rust/ockam/ockam_transport_core/src/lib.rs#L1-L7

And introduces no new dependencies

https://github.com/ockam-network/ockam/blob/c9f06882c0af1d336d65e3da575438717254b26a/implementations/rust/ockam/ockam_transport_core/Cargo.toml#L41-L44

**Comment by SanjoDeundiak** [comment]
I don't see a reason why we should move things **_out_** of that crate, the plan was to move things **_to_** that crate, like common code in TCP and WS. We're just not there yet.

**Comment by thomcc** [comment]
Yes, I think this is okay for now. I still don't quite know what the abstraction we want here is for transports, but I think this crate probably should exist long term. One of the things it will help with is the abstraction for the async runtime, since the IO impls are typically integrated with it (e.g. `tokio::net::*` depends on tokio). I haven't done all the design work for this, but I suspect this will be where some stuff along these lines lives...

I don't mind removing it for now if we want to re-add it later, though.
